.ci/build.sh

```diff
@@ -5,9 +5,37 @@ SCRIPT_DIR="$(dirname "$0")"
 set -exuo pipefail
 source "$SCRIPT_DIR/common"
 
-docker build -t $CI_IMAGE_URI --build-arg IMAGE_REPOSITORY=${IMAGE} --build-arg IMAGE_TAG=${COMMIT} "$SCRIPT_DIR/.."
+command -v docker &>/dev/null || ( echo 'Docker is not installed. Aborting.'; exit 1 )
+docker buildx version &>/dev/null || ( echo 'Docker Buildx plugin is not installed. Aborting.'; exit 1 )
 
-if [ -n "${DNAME:-}" ] && [ -n "${DPASS:-}" ]; then
-  docker login -u "${DNAME}" -p "${DPASS}";
-  TagAndPushImage $CI_IMAGE_REPO $CI_TAG;
+export PUSH_OPTION=""
+export IMAGE_TAGS="${CI_TEST_IMAGE_TAG}"
+
+if [ "${CI_IMAGE_PLATFORMS}" = "local" ]; then
+  export IMAGE_DESTINATION="docker"
+else
+  export IMAGE_DESTINATION="registry"
+  if [ -n "${DNAME:-}" ] && [ -n "${DPASS:-}" ]; then
+    docker login -u "${DNAME}" -p "${DPASS}"
+    export PUSH_OPTION="--push"
+  fi
+  export IMAGE_TAGS="${COMMIT} ${BRANCH_SLUG}";
+fi
+
+for t in ${IMAGE_TAGS}; do
+  TAGS_ARGS+=" -t $(build-image-uri ${t})"
+done
+
+docker buildx build \
+  --pull --no-cache ${PUSH_OPTION} \
```

Comment:
Should we skip pull and nocache for local builds? would make iterative local dev and deploy faster

Reply:
Will make them conditional too.